msg.go

```diff
@@ -427,10 +427,7 @@ Loop:
 		off1 = off
 	}
 	if len(s) == 0 {
-		s = []byte(".")
-	} else if len(s) >= maxLen {
-		// error if the name is too long, but don't throw it away
-		return string(s), lenmsg, ErrLongDomain
+		return ".", off1, nil
 	}
 	return string(s), off1, nil
```

Comment:
```suggestion
	return *(*string)(unsafe.Pointer(&s)), off1, nil
```

another string cast costly, double allocate here, with this change can stop another allocation. This function from `strings.Builder` and discussed here [https://github.com/golang/go/issues/25484#issuecomment-391415660](url)

Reply:
there is probably lower hanging fruit (elsewhere) than to micro optimize these allocations. This is borderline unreadable...

Reply:
@semihalev That's out of scope for this PR. I actually suspect that change would make things worse in this function because I think `s := make([]byte, 0, 64)` is being stack allocated in most cases. Whereas if you were to make that change, it would cause `s` to escape and be heap allocated.

I've tried switching this (`UnpackDomainName`) over to `strings.Builder` several times now, but always seem to end up with worse performance. (You can see 5debfeec63c371d5806bb64f4553078717b2eef5 & 0d29b283ac0f967dd3a02739bf26a22702210d7a where I've switched other parts of the code to use `strings.Builder`).